### PR TITLE
Create baseline event field and event template annotations

### PR DESCRIPTION
**Pull request by chipingyeh**
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)

### CODE REVIEW

**Review by chipingyeh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: 0 of 1 files reviewed, 1 unresolved discussion

*[a discussion](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtP_MjYD7g1L3mQPP6H:-MtP_MjYD7g1L3mQPP6I:bsgb9tu) (no related file):*
This is more or less taken straight from the design doc -- please let me know if there's anything I'm missing.

---

**Review by robinsons — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @chipingyeh and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 17 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiIqOkBVMKrDFYC7L8:-MtiIqOkBVMKrDFYC7L9:bwimtqd) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L17)):*
> ```protobuf
> syntax = "proto3";
> 
> package wfa.measurement.internal.kingdom;
> ```

The `package` should be in the public API, `wfa.measurement.api.v2alpha`, rather than internal.
___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 19 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJ9G6D-RdVDg_M4f9:-MtiJ9G6D-RdVDg_M4fA:bupdsss) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L19)):*
> ```protobuf
> package wfa.measurement.internal.kingdom;
> 
> option java_package = "org.wfanet.measurement.internal.kingdom";
> ```

The `java_package` should be `org.wfanet.measurement.api.v2alpha`.
___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 24 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJXs77iZqqXgHs2dq:-MtiJXs77iZqqXgHs2dr:bgvuyo) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L24)):*
> ```protobuf
> // Metadata about an Event Field.
> message EventFieldDescriptor {
>   string display_name = 1;
> ```

As part of the public API, each field should have a comment (even if it seems obvious what the field is) and indicate whether it's required or not. At the moment we can't really enforce required fields in any automatic way, but in the future we might create tooling to help with template creation, and that tooling could do the enforcement.

As an example, for `display_name` something like: "Human-readable name of the Event Field, e.g. for display in a UI. Required."
___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 41 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiGeJ68Yp_D9vFjX3C:-MtiGeJ74HgNavga_-TB:bhhp0eq) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L41)):*
> ```protobuf
> // Custom options for message fields.
> extend google.protobuf.FieldOptions {
>   EventFieldDescriptor event_field = 50001;
> ```

I believe we also decided to adopt the variant described [here](https://docs.google.com/document/d/1SEHbHhWz3p1_E1keSrlbbW1Z9nEPClCrS9bjYWgilTo/edit?pli=1#bookmark=id.7bmi6ocdkna4) -- specifically, making top-level "wrapper" messages for each event field type. This would make the `EventFieldDescriptor` an extension on `MessageOptions` rather than `FieldOptions`.

**Review by chipingyeh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: 0 of 1 files reviewed, 5 unresolved discussions (waiting on @chipingyeh, @robinsons, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 17 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiIqOkBVMKrDFYC7L8:-Mtic1VR43Sg39C5QpE7:b78s84j) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L17)):*
> wfa.measurement.api.v2alpha
Good catch -- done.
___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 19 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJ9G6D-RdVDg_M4f9:-Mtic7fsC4C4cgr9wtZ6:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L19)):*
<details><summary><i>Previously, robinsons wrote…</i></summary><blockquote>

The `java_package` should be `org.wfanet.measurement.api.v2alpha`.
</blockquote></details>

Done.
___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 24 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJXs77iZqqXgHs2dq:-MticZcLDlBpDzkCLRTg:b-8rq403) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L24)):*
<details><summary><i>Previously, robinsons wrote…</i></summary><blockquote>

As part of the public API, each field should have a comment (even if it seems obvious what the field is) and indicate whether it's required or not. At the moment we can't really enforce required fields in any automatic way, but in the future we might create tooling to help with template creation, and that tooling could do the enforcement.

As an example, for `display_name` something like: "Human-readable name of the Event Field, e.g. for display in a UI. Required."
</blockquote></details>

Updated fields with basic comments -- feel free to suggest any edits.
___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 41 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiGeJ68Yp_D9vFjX3C:-MticX0L0-YQpoQ9nv1y:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L41)):*
<details><summary><i>Previously, robinsons wrote…</i></summary><blockquote>

I believe we also decided to adopt the variant described [here](https://docs.google.com/document/d/1SEHbHhWz3p1_E1keSrlbbW1Z9nEPClCrS9bjYWgilTo/edit?pli=1#bookmark=id.7bmi6ocdkna4) -- specifically, making top-level "wrapper" messages for each event field type. This would make the `EventFieldDescriptor` an extension on `MessageOptions` rather than `FieldOptions`.
</blockquote></details>

Done.

**Review by robinsons — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 2 unresolved discussions (waiting on @chipingyeh, @SanjayVas, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 19 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJ9G6D-RdVDg_M4f9:-MtjUX0A-JYrACG3_TOy:b76myds) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L19)):*
<details><summary><i>Previously, chipingyeh wrote…</i></summary><blockquote>

Done.
</blockquote></details>

Note that the proto package (`package`) and the `java_package` are different. Can you check again?

**Review by robinsons — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 2 unresolved discussions (waiting on @chipingyeh, @SanjayVas, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 19 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJ9G6D-RdVDg_M4f9:-MtjV6FrCY635aAaFnqC:b-qy6eg3) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L19)):*
<details><summary><i>Previously, robinsons wrote…</i></summary><blockquote>

Note that the proto package (`package`) and the `java_package` are different. Can you check again?
</blockquote></details>

That is to say, they *should be* different, but they're currently the same.

**Review by chipingyeh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: 0 of 1 files reviewed, 2 unresolved discussions (waiting on @chipingyeh, @robinsons, @SanjayVas, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 19 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJ9G6D-RdVDg_M4f9:-MtnCSWU1ZJwHwKQi9Ri:b-urcap) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L19)):*
<details><summary><i>Previously, robinsons wrote…</i></summary><blockquote>

That is to say, they *should be* different, but they're currently the same.
</blockquote></details>

Ah, I see -- done. This is so the generated files are in the same package as the Kotlin files, while the proto files are in their own package?

**Review by SanjayVas — APPROVED** [review]
> Reviewed 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 2 unresolved discussions (waiting on @chipingyeh, @robinsons, and @stevenwarejones)

**Review by SanjayVas — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 3 unresolved discussions (waiting on @chipingyeh, @robinsons, and @stevenwarejones)

*[a discussion](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtnSpF022OKHu-yJsGD:-MtnSpF022OKHu-yJsGE:b-nat0h5) (no related file):*
Sorry, prematurely approved. There should be matching target(s) in the BUILD.bazel file.

---

**Review by robinsons — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 2 unresolved discussions (waiting on @chipingyeh, @SanjayVas, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 19 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtiJ9G6D-RdVDg_M4f9:-Mto-vLaAqSBcNNNiNS1:b-htdvlo) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/a0eca8a7374e8043213aed338a5573dd60374c90/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L19)):*
<details><summary><i>Previously, chipingyeh wrote…</i></summary><blockquote>

Ah, I see -- done. This is so the generated files are in the same package as the Kotlin files, while the proto files are in their own package?
</blockquote></details>

Basically. The `package` declaration is what you'd use if you were to import this file into another `.proto` file, e.g. `import "wfa/measurement/api/v2alpha/event_annotations.proto";`. The `java_package` is what you'd use in a Java/Kotlin `import` statement. It's also the output package for generated `.class` files.

**Review by stevenwarejones — COMMENTED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 3 unresolved discussions (waiting on @chipingyeh and @SanjayVas)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 42 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtySn_B03W8elp3It3E:-MtySn_B03W8elp3It3F:bpzq9nk) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/c77c8bcd33743e10efeaea220e40a50256715d84/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L42)):*
> ```protobuf
> // Custom options for messages.
> extend google.protobuf.MessageOptions {
>   EventTemplateDescriptor event_template = 50000;
> ```

why are we starting at 50000?

**Review by chipingyeh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 3 unresolved discussions (waiting on @chipingyeh, @SanjayVas, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 42 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtySn_B03W8elp3It3E:-MtyVGdP1wTfPxYPyo59:b-d98wpu) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/c77c8bcd33743e10efeaea220e40a50256715d84/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L42)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

why are we starting at 50000?
</blockquote></details>

This was referenced from the [online documentation](https://developers.google.com/protocol-buffers/docs/proto#customoptions)

**Review by SanjayVas — APPROVED** [review]
> Reviewed 2 of 2 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 2 unresolved discussions (waiting on @chipingyeh and @stevenwarejones)

**Review by stevenwarejones — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: all files reviewed, 2 unresolved discussions (waiting on @chipingyeh)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 42 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtySn_B03W8elp3It3E:-MuBtg-F-0SIiq-5xwSJ:buxxn3d) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/c77c8bcd33743e10efeaea220e40a50256715d84/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L42)):*
<details><summary><i>Previously, chipingyeh wrote…</i></summary><blockquote>

This was referenced from the [online documentation](https://developers.google.com/protocol-buffers/docs/proto#customoptions) 


</blockquote></details>

can you add that as a comment?

**Review by chipingyeh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: 1 of 2 files reviewed, 2 unresolved discussions (waiting on @chipingyeh, @robinsons, @SanjayVas, and @stevenwarejones)

___
*[src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto, line 42 at r4](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84#-MtySn_B03W8elp3It3E:-MuBwHgb9pJOFJZpjWnq:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement-api/blob/c77c8bcd33743e10efeaea220e40a50256715d84/src/main/proto/wfa/measurement/api/v2alpha/event_annotations.proto#L42)):*
<details><summary><i>Previously, stevenwarejones (Steven Ware Jones) wrote…</i></summary><blockquote>

can you add that as a comment?
</blockquote></details>

Done.

**Review by stevenwarejones — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement-api/84)* status: 1 of 2 files reviewed, 1 unresolved discussion (waiting on @chipingyeh, @robinsons, @SanjayVas, and @stevenwarejones)
